lib/order.rb

```diff
@@ -0,0 +1,118 @@
+require_relative 'customer'
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  @@customer_orders = []
+
+  VALID_FULFILLMENTS = [:pending, :paid, :processing, :shipped, :complete]
+
+  def initialize(id, products, customer,  fulfillment_status = :pending)
+
+    if !VALID_FULFILLMENTS.include?(fulfillment_status)
+
+      raise ArgumentError.new("Not a valid status")
+    end
+
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+  end
+
+  def total
+    sum = @products.reduce(0) do |total,(item, cost)|
```

Comment:
Fantaaaastic, but the syntax here looks a little weird. What's up with `total,(item, cost)`?